spec/services/send_welcome_email_service_spec.rb

```diff
@@ -51,12 +53,41 @@
       expect(current_user_spy).not_to have_received(:update).with(hash_including(first_login_date_utc: Time.now.utc))
     end
 
-    it 'does not update their welcome email date' do
-      expect(current_user_spy).not_to have_received(:update).with(hash_including(welcome_email_date_utc: Time.now.utc))
+    context 'And has recieved a welcome email' do
+      it 'does not update their welcome email date' do
+        expect(current_user_spy).not_to have_received(:update).with(hash_including(welcome_email_date_utc: Time.now.utc))
+      end
+
+      it 'does not send the welcome email' do
+        expect(mailer_spy).not_to have_received(:send_welcome_email)
+      end
     end
 
-    it 'does not send the welcome email' do
-      expect(mailer_spy).not_to have_received(:send_welcome_email)
+    context 'And has not recieved a welcome email' do
```

Comment:
```suggestion
    context 'And has not received a welcome email' do
```
i before e only when you can see